kadi/update_events.py

```diff
@@ -149,11 +183,11 @@ def update(EventModel, date_stop):
                     foreign_model = ForeignModel.from_dict(row, logger)
                     setattr(foreign_model, event_model.model_name, event_model)
                     logger.verbose('Adding {}'.format(foreign_model))
-                    foreign_model.save()
+                    try4times(foreign_model.save)
 
-    # If processing got here with no exceptions then save the event update
-    # information to database
-    update.save()
+        # If processing got here with no exceptions then save the event update
```

Comment:
It seems that this final `update.save()` should be in the atomic commit block, because it too can fail due to DB locked.  That's not good because you have the events being updated without the last date of update being correspondingly updated.  (Though I think in normal circumstance it gets cleaned up the next time around with no problems).